src/D2L.CodeStyle.Analyzers/UnsafeStaticsAnalyzer.cs

```diff
@@ -49,6 +49,11 @@ public sealed class UnsafeStaticsAnalyzer : DiagnosticAnalyzer {
         }
 
         private void AnalyzeField( SyntaxNodeAnalysisContext context ) {
+            if( context.Node.SyntaxTree.FilePath.Contains( ".generated" ) ) {

```

Comment:
Yes, `.generated` is a folder name, not an extension, so it can be anywhere in the path.